### PR TITLE
Specify constraints on metadata member

### LINKED ISSUE #5: Describe the metadata dictionary's typing constraints in SerializableObjet
All SerializableObjects have a metadata dictionary.
These dictionaries adhere to typing constraints very similar to json: 

- all keys must be strings
- values can be: string, integer, double, dictionary, array, boolean, null
- values can also be other SerializableObjects

### PR DESCRIPTION
**Pull request by palemieux**
Closes #5

### CODE REVIEW

File: src/main/md/otio-core.md [hunk 1]
```diff
@@ -1775,11 +1775,21 @@ Double (binary64) floating point number as defined in IEEE 754.
...
+
+A `MetadataObject` instance and any of its descendent `object` values may be an instance of
+[`SerializableObject`](#object-model-SerializableObject).
+
+_EXAMPLE_: A `MetadataObject` instance can contain `RationalTime` and `TimeRange` instances.
+
+#### Processing
+
+Unless it is a descendent of an instance of [`SerializableObject`](#object-model-SerializableObject), a processor shall interpret a
+`number` value as a [`Float`](#object-model-Float).
```
Comment (reinecke) [inline src/main/md/otio-core.md]: I'm not sure I fully understand this. Is this saying all numbers should be interpreted as float values unless they are part of a `SerializableObject`?

**Review by reinecke — COMMENTED** [review]
This is a good clarification.

File: src/main/md/otio-core.md (hunk 1 above)
Comment (palemieux) [reply src/main/md/otio-core.md]: `Float` type in the specification is defined as a IEEE double. I can see how this might be confusing. What about changing the name of the `Float` type to `Double`?

Comment (palemieux) [reply src/main/md/otio-core.md]: @reinecke The idea is that the interpretation of _JSON numbers_ as used by `SerializableObject` and its descendants, is determined by the object model, e.g. they are sometimes interpreted as integers. For other _JSON numbers_ not used by `SerializableObject` and its descendants, they are interpreted as _IEEE double_. Makes sense?

Comment (reinecke) [reply src/main/md/otio-core.md]: Okay, that makes sense. I think I misinterpreted the context as being part of metadata object when it is a separate section alltogether.

On the point of float vs. double: I agree about switching to Double. This matches the historical context best.

Comment (palemieux) [reply src/main/md/otio-core.md]: @reinecke Changed floating point type name to `Double`

Comment (reinecke) [reply src/main/md/otio-core.md]: I was just reviewing one more time and the sentence still feels hard to parse the meaning of. Perhaps we could try wording like:
```
A processor shall interpret a `number` value as a [`Double`](#object-model-Float) unless it is a descendent of an instance of [`SerializableObject`](#object-model-SerializableObject) and the schema dictates a different interpretation.
```

One other detail is that in the OTIO implementations, numbers without the fraction (`.`) or exponent (`e`|`E`) delimiters are interpreted as `int`. Is this something we should call out here?

Comment (palemieux) [reply src/main/md/otio-core.md]: > One other detail is that in the OTIO implementations, numbers without the fraction (.) or exponent (e|E) delimiters are interpreted as int. Is this something we should call out here?

Is this a requirement of all implementations? If so, it needs to be called out.

Comment (reinecke) [reply src/main/md/otio-core.md]: I can't think of a reason it has to work this way, it's really up to the consumer what representation is in memory I suppose.

**Review by reinecke — APPROVED** [review]
